lms/views/predicates/_lti_launch.py

```diff
@@ -71,34 +130,38 @@ def blackboard_course_copied_assignment_launch_view(context, request):
     """
 
     name = "blackboard_copied"
+    param_name = "resource_link_id_history"
 
-    def __init__(self, value, config):
-        super().__init__(value, config)
-        self.db_configured = DBConfigured(True, config)
 
-    def __call__(self, context, request):
-        if self.db_configured(context, request):
-            # We already have a document URL in the DB for this resource_link_id,
-            # so it's not a newly copied assignment.
-            is_newly_copied = False
-        else:
-            resource_link_id_history = request.params.get("resource_link_id_history")
-            tool_consumer_instance_guid = request.params.get(
-                "tool_consumer_instance_guid"
-            )
+class BrightspaceCopied(_CourseCopied):
+    """
+    Allow invoking an LTI launch view for newly course-copied Brightspace assignments.
 
-            if not resource_link_id_history or not tool_consumer_instance_guid:
-                is_newly_copied = False
-            else:
-                # Look for the document URL of the previous assignment that
-                # this one was copied from.
-                assignment_service = request.find_service(name="assignment")
-                previous_document_url = assignment_service.get_document_url(
-                    tool_consumer_instance_guid, resource_link_id_history
-                )
-                is_newly_copied = bool(previous_document_url)
+    When a user uses Brightspace's Course Copy feature to copy a course that
+    contains Hypothesis assignments, Brightspace copies the Hypothesis
+    assignments into the new course. Brightspace gives new resource_link_id's
+    to the new copies of the assignments and includes the original
+    resource_link_id in an ext_d2l_resource_link_id_history launch param.
 
-        return is_newly_copied == self.value
+    Pass brightspace_copied=True to a view's configuration to allow invoking
+    the view only for newly course-copied Brightspace assignments where we
+    don't yet have a document_url for the new resource_link_id but we do have a
+    document_url for the ext_d2l_resource_link_id_history.
+
+    For example:
+
+        @view_config(..., brightspace_copied=True)
+        def brightspace_course_copied_assignment_launch_view(context, request):
+            ...
+
+    ext_d2l_resource_link_id_history is a non-standard and undocumented param
+    that's only used by Brightspace, but nothing in this code actually prevents
+    us from using ext_d2l_resource_link_id_history if another LMS sends it to
+    us.
```

Comment:
I think these comments are a kind of code repetition / maintainance problem. The base class could probably do with a description of what it is as well.

We can probably have our cake and eat it by genericising this a bit and adding it to the base class, and then replacing this and the Blackboard ones with something like:

```
Brightspace custom parameter based course copy detection (see base class for details)

Usage:

        @view_config(..., brightspace_copied=True)
        def brightspace_course_copied_assignment_launch_view(context, request):
            ...